nrf-hal-common/src/spim.rs

```diff
@@ -361,8 +390,41 @@ where
     }
 
     /// Return the raw interface to the underlying SPIM peripheral.
-    pub fn free(self) -> T {
-        self.0
+    pub fn free(self) -> (T, Pins) {
+        (self.periph, self.pins)
+    }
+
+    pub fn dma_transfer_split<TxW, RxW, TxB, RxB>(
+        mut self,
+        tx_buffer: TxB,
+        mut rx_buffer: RxB,
+    ) -> Result<TransferSplit<T, TxB, RxB>, (Self, Error)>
+    where
+        TxB: ReadBuffer<Word = TxW>,
```

Comment:
```suggestion
        TxB: ReadBuffer<Word = TxW> + 'static,
```
Based on the matrix channel discussion today it looks like we need + 'static bounds on these buffers aswell